src/erlcloud_sns.erl

```diff
@@ -387,45 +389,45 @@ list_platform_applications(NextToken, Config) ->
 list_platform_applications(NextToken, AccessKeyID, SecretAccessKey) ->
     list_platform_applications(NextToken, new_config(AccessKeyID, SecretAccessKey)).
 
 
 
--spec publish_to_topic/2 :: (string(), sns_message()) -> string().
+-spec publish_to_topic/2 :: (parameter(), sns_message()) -> string().
 publish_to_topic(TopicArn, Message) ->
     publish_to_topic(TopicArn, Message, undefined).
 
--spec publish_to_topic/3 :: (string(), sns_message(), undefined|string()) -> string().
+-spec publish_to_topic/3 :: (parameter(), sns_message(), undefined|parameter()) -> string().
 publish_to_topic(TopicArn, Message, Subject) ->
     publish_to_topic(TopicArn, Message, Subject, default_config()).
 
--spec publish_to_topic/4 :: (string(), sns_message(), undefined|string(), aws_config()) -> string().
+-spec publish_to_topic/4 :: (parameter(), sns_message(), undefined|parameter(), aws_config()) -> string().
 publish_to_topic(TopicArn, Message, Subject, Config) ->
     publish(topic, TopicArn, Message, Subject, Config).
 
--spec publish_to_topic/5 :: (string(), sns_message(), undefined|string(), string(), string()) -> string().
+-spec publish_to_topic/5 :: (parameter(), sns_message(), undefined|parameter(), parameter(), parameter()) -> string().
 publish_to_topic(TopicArn, Message, Subject, AccessKeyID, SecretAccessKey) ->
     publish_to_topic(TopicArn, Message, Subject, new_config(AccessKeyID, SecretAccessKey)).
 
--spec publish_to_target/2 :: (string(), sns_message()) -> string().
+-spec publish_to_target/2 :: (parameter(), sns_message()) -> string().
 publish_to_target(TargetArn, Message) ->
     publish_to_target(TargetArn, Message, undefined).
 
--spec publish_to_target/3 :: (string(), sns_message(), undefined|string()) -> string().
+-spec publish_to_target/3 :: (parameter(), sns_message(), undefined|parameter()) -> string().
 publish_to_target(TargetArn, Message, Subject) ->
     publish_to_target(TargetArn, Message, Subject, default_config()).
 
--spec publish_to_target/4 :: (string(), sns_message(), undefined|string(), aws_config()) -> string().
+-spec publish_to_target/4 :: (parameter(), sns_message(), undefined|parameter(), aws_config()) -> string().
 publish_to_target(TargetArn, Message, Subject, Config) ->
     publish(target, TargetArn, Message, Subject, Config).
 
--spec publish_to_target/5 :: (string(), sns_message(), undefined|string(), string(), string()) -> string().
+-spec publish_to_target/5 :: (parameter(), sns_message(), undefined|parameter(), parameter(), parameter()) -> string().
 publish_to_target(TargetArn, Message, Subject, AccessKeyID, SecretAccessKey) ->
     publish_to_target(TargetArn, Message, Subject, new_config(AccessKeyID, SecretAccessKey)).
 
--spec publish/5 :: (topic|target, string(), sns_message(), undefined|string(), aws_config()) -> string().
+-spec publish/5 :: (topic|target, parameter(), sns_message(), undefined|parameter(), aws_config()) -> string().
```

Comment:
Line 430 is too long: -spec publish/5 :: (topic|target, parameter(), sns_message(), undefined|parameter(), aws_config()) -> string()..